### ISSUE TITLE
ninja: warning: multiple rules generate lib/nvim.lib

### ISSUE THREAD

**Issue by drichardson** [neovim/neovim#12115]
As of commit 46fdad255ead9ca484a6e512efe13b379b8fc8ab, ninja reports the following warning when building:

- commit 46fdad255ead9ca484a6e512efe13b379b8fc8ab
- build environment: x64 Native Tools Command Prompt for VS 2019, which includes ninja 1.8.2 and cmake 3.15.19101501-MSVC_2
- Operating system/version: Windows 10

### Steps to reproduce

```
mkdir build
cd build
cmake -G Ninja ..
ninja
```

### Actual behaviour

```
ninja: warning: multiple rules generate lib/nvim.lib. builds involving this target will not be correct; continuing anyway
[-w dupbuild=warn]
```

### Expected behaviour

No warning.

### Notes

build.ninja contains the the following:

```
build bin\nvim.exe lib\nvim.lib: C_EXECUTABLE_LINKER__nvim...
...
build lib\nvim.lib: C_STATIC_LIBRARY_LINKER__libnvim s...
```

This shows why ninja is complaining (two build rules have lib\nvim.lib in the list of outputs). However, it is not immediately clear to me what in the CMakeLists.txt is causing it.

**Comment by drichardson** [comment]
I think this has something to do with the following line in src/nvim/CMakeLists.txt:

```
set_target_properties(
  libnvim
  PROPERTIES
    POSITION_INDEPENDENT_CODE ON
    OUTPUT_NAME nvim
)
```

If I change the OUTPUT_NAME to something else, the warning does not occur. It seems like the cmake Ninja generator is getting confused here and using the same build rule for both nvim.exe and libnvim when OUTPUT_NAME is set to nvim.

**Comment by teto** [comment]
the mentioned commit doesn't touch cmake so I suspect the cause to be external. If you submit a PR changing the OUTPUT_NAME of libnvim to let's say libnvim, we can see how the CI reacts and if the message goesaway.

**Comment by drichardson** [comment]
> the mentioned commit doesn't touch cmake

Oh sorry, I wasn't trying to suggest this. I was just saying that the warning occurred at that commit (and probably at an earlier one as well). I did not try to find the initial commit that caused the warning.

> so I suspect the cause to be external.

I agree. It seems like cmake is incorrectly using nvim.exe's build rule due to the output names matching.

> If you submit a PR changing the OUTPUT_NAME of libnvim to let's say libnvim, we can see how the CI reacts and if the message goesaway.

It does go away, I already tried that. Due to my lack of cmake knowledge, I was concerened that this may cause a problem, though I didn't notice any problem in my testing.
